usr/src/uts/common/io/scsi/adapters/smartpqi/smartpqi.h

```diff
@@ -0,0 +1,627 @@
+#include <sys/sysevent/dr.h>
+#include <sys/sata/sata_defs.h>
+#include <sys/sata/sata_hba.h>
+#include <sys/scsi/generic/sas.h>
+#include <sys/scsi/impl/scsi_sas.h>
+#include <sys/scsi/impl/scsi_reset_notify.h>
+#include <sys/sdt.h>
+#include <sys/mdi_impldefs.h>
+#include <sys/fs/dv_node.h>
+#include <sys/sysmacros.h>
+#include <sys/systm.h>
+#include <smartpqi_hw.h>
+
+/* ---- Hint for ddi_soft_state_init() on amount of structs to alloc ---- */
+#define	SMARTPQI_INITIAL_SOFT_SPACE	1
+
+#define	SMARTPQI_MOD_STRING	"smartpqi 20180525"
+
+/* ---- Handy constants ---- */
+#define	UNDEFINED				-1
+#define	MAX_NAME_PROP_SIZE			256
+#define	LUN_PROP				"lun"
+#define	LUN64_PROP				"lun64"
+#define	MDI_GUID				"wwn"
+#define	NDI_GUID				"guid"
+#define	TARGET_PROP				"target"
+#define	LUN_PROP				"lun"
+#define	COMPAT_PROP				"compatible"
+#define	NAME_DISK				"disk"
+#define	NAME_ENCLOSURE				"enclosure"
+
+#define	CMD_TIMEOUT_SCAN_SECS			10
+#define	SYNC_CMDS_TIMEOUT_SECS			5
+#define	IO_SPACE				1
+#define	PQI_MAXTGTS				256
+
+#define	PQI_MIN_MSIX_VECTORS			1
+#define	PQI_MAX_MSIX_VECTORS			16
+#define	PQI_DEFAULT_QUEUE_GROUP			0
+#define	PQI_MAX_QUEUE_GROUPS			PQI_MAX_MSIX_VECTORS
+#define	PQI_MIN_OPERATIONAL_QUEUE_ID		1
+/* ---- Size of structure scsi_arq_status without sense data. ---- */
+#define PQI_ARQ_STATUS_NOSENSE_LEN    (sizeof (struct scsi_arq_status) - \
+    sizeof (struct scsi_extended_sense))
+
+/* ---- macros to return various addresses ---- */
+#define	ADDR2TRAN(ap)	((ap)->a_hba_tran)
+#define	TRAN2PQI(hba)	((pqi_state_t)(hba)->tran_hba_private)
+#define	ADDR2PQI(ap)	(TRAN2PQI(ADDR2TRAN(ap)))
+#define	PKT2CMD(pkt)	((pqi_cmd_t)(pkt)->pkt_ha_private)
+#define	CMD2PKT(cmd)	((struct scsi_pkt *)(cmd)->pc_pkt)
+
+/* ---- PQI configuration ---- */
+#define	PQI_MAX_OUTSTANDING_REQUESTS		32
+#define	PQI_ERROR_BUFFER_ELEMENT_LENGTH sizeof (struct pqi_raid_error_info)
+#define	PQI_CREATE_ADMIN_QUEUE_PAIR		1
+#define	PQI_DELETE_ADMIN_QUEUE_PAIR		2
+#define	PQI_MAX_TRANSFER_SIZE			(4 * 1024U * 1024U)
+#define	PQI_MAX_RESET_XFER_SIZE			(512 * 1024)
+#define	PQI_NUM_SUPPORTED_EVENTS		7
+#define	PQI_RESERVED_IO_SLOTS_LUN_RESET		1
+#define	PQI_RESERVED_IO_SLOTS_EVENT_ACK	PQI_NUM_SUPPORTED_EVENTS
+#define	PQI_RESERVED_IO_SLOTS_SYNCHRONOUS_REQUESTS	3
+#define	PQI_RESERVED_IO_SLOTS \
+	(PQI_RESERVED_IO_SLOTS_LUN_RESET + PQI_RESERVED_IO_SLOTS_EVENT_ACK + \
+	PQI_RESERVED_IO_SLOTS_SYNCHRONOUS_REQUESTS)
+#define	PQI_MAX_SCATTER_GATHER			0x200
+
+/* ---- SIS constants ---- */
+#define	SIS_BASE_STRUCT_ALIGNMENT		16
+
+/* ---- Once every 10 seconds ---- */
+#define	WATCHDOG (10 * MICROSEC)
+
+/* ---- Update HBA time of day clock once a day ---- */
+#define	MINUTE		60
+#define	HOUR		(60 * MINUTE)
+#define	DAY		(24 * HOUR)
+
+#define	HBA_IS_QUIESCED(s)	(((s)->s_flags & PQI_HBA_QUIESCED) != 0)
+#define	HBA_QUIESCED_PENDING(s)	\
+	(((s)->s_flags & PQI_HBA_QUIESCED_PENDING) != 0 && \
+	((s)->s_cmd_queue_len == 0))
+#define	PQIALIGN_TYPED(addr, align, type) \
```

Comment:
Seems like this should use the sysmacros.h one and just cast the result in code? (I don't see the advantage to the cast being in the parameters).

If there _is_ an advantage, it should be defined in terms of the one in sysmacros.h regardless.